libtenzir/builtins/operators/decapsulate.cpp

```diff
@@ -268,109 +281,113 @@ auto parse(record_ref builder, std::span<const std::byte> bytes,
   return std::nullopt;
 }
 
-struct operator_args {
-  std::optional<located<uint16_t>> vxlan_port;
-
-  friend auto inspect(auto& f, operator_args& x) -> bool {
-    return f.object(x)
-      .pretty_name("operator_args")
-      .fields(f.field("vxlan_port", x.vxlan_port));
+auto decapsulate(const series& s, diagnostic_handler& dh,
+                 bool include_old) -> std::optional<series> {
+  // Get the packet payload.
+  if (s.type.kind().is_not<record_type>()) {
+    if (s.type.kind().is_not<null_type>()) {
+      diagnostic::warning("cannot decapsulate type `{}`", s.type).emit(dh);
```

Comment:
```suggestion
      diagnostic::warning("expected `record`, got `{}`", s.type.kind()).emit(dh);
```

A bit more helpful, and `.kind()` in case `s.type` is a huge, nested type definition.